AZ3166/src/libraries/MQTT/src/MQTTNetwork.h

```diff
@@ -9,46 +9,92 @@
 #include "SystemWiFi.h"
 #include "Telemetry.h"
 
-class MQTTNetwork {
-public:
-    MQTTNetwork() {
-        socket = new TCPSocket();
+class MQTTNetwork
+{
+  public:
+    MQTTNetwork()
+    {
+        _tcpSocket = NULL;
     }
 
-    ~MQTTNetwork() {
-        delete socket;
+    ~MQTTNetwork()
+    {
+        if (_tcpSocket != NULL)
+        {
+            _tcpSocket->close();
+            delete _tcpSocket;
+            _tcpSocket = NULL;
+        }
     }
 
-    int read(unsigned char* buffer, int len, int timeout) {
-        return socket->recv(buffer, len);
+    int read(unsigned char *buffer, int len, int timeout)
+    {
+        return _tcpSocket->recv(buffer, len);
```

Comment:
if not null {return recv}
return 0;

Reply:
return NSAPI_ERROR_NO_SOCKET when TCPSocket is not availible, otherwise return the socket receive result.

---
In reply to: [182332085](https://github.com/Microsoft/devkit-sdk/pull/809#discussion_r182332085) [](ancestors = 182332085)